inst/include/Rcpp/sugar/functions/sapply.h

```diff
@@ -40,9 +39,10 @@ struct sapply_application_result_of
         typedef typename ::std::invoke_result<Function, typename SugarExpression::stored_type>::type type;
     #endif
 #else
+    // TODO this else branch can likely go
```

Comment:
Yes, I think so.

Reply:
I left it on purpose because result_of() is used in a number of places, and there is an implementation file supporting it so removal is another subtask.